topo_processor/metadata/metadata_validators/metadata_validator_tiff.py

```diff
@@ -27,44 +25,37 @@ def is_applicable(self, item: Item) -> bool:
-                    with rasterio.open(f) as tiff:
-                        # black and white
-                        if ColorInterp.gray in tiff.colorinterp and len(tiff.colorinterp) == 1:
-                            # check linz_geospatial_type matches colorinterp
-                            if geospatial_type not in ["black and white image", "black and white infrared image"]:
-                                raise Exception(
-                                    f"Wrong linz_geospatial_type of '{geospatial_type}' when bands = {', '.join([color.name for color in tiff.colorinterp])}"
-                                )
-                            # check eo:bands matches colorinterp
-                            if len(eo_bands) != 1 or eo_bands[0]["common_name"] != "pan":
-                                raise Exception(
-                                    f"Wrong 'eo:bands' with common name: '{geospatial_type}' when bands = {', '.join([color.name for color in tiff.colorinterp])}"
-                                )
-                        # color
-                        if all(band in [ColorInterp.red, ColorInterp.blue, ColorInterp.green] for band in tiff.colorinterp):
-                            common_names = [common_names["common_name"] for common_names in eo_bands]
-                            # check linz_geospatial_type matches colorinterp
-                            if geospatial_type not in ["color image", "color infrared image"]:
-                                raise Exception(
-                                    f"Wrong linz_geospatial_type of '{geospatial_type}' when bands = {', '.join([color.name for color in tiff.colorinterp])}"
-                                )
-                            # check eo:bands matches colorinterp
-                            if (
-                                len(eo_bands) != 3
-                                or "red" not in common_names
-                                or "green" not in common_names
-                                or "blue" not in common_names
-                            ):
-                                raise Exception(
-                                    f"Wrong 'eo:bands' with common name: {geospatial_type} when bands = {', '.join([color.name for color in tiff.colorinterp])}"
-                                )
-                for warn in w:
-                    get_log().warning(f"Rasterio Warning: {warn.message}", file=asset.source_path, loader=self.name)
+
+            with warnings.catch_warnings(record=True) as w:
+
+                # black and white
+                if geospatial_type in ["black and white image", "black and white infrared image"]:
+                    # check eo:bands matches geospatial_type
+                    if len(eo_bands) != 1 or eo_bands[0]["common_name"] != "pan":
+                        raise Exception(f"Wrong linz_geospatial_type of '{geospatial_type}' when bands = '{eo_bands}'")
+                    else:
```

Comment:
what does the continue do here?

Reply:
Sorry, I rewrote it and forgot to remove them.

Reply:
Might be better now :sweat_smile: